src/script/descriptor.cpp

```diff
@@ -1464,6 +1467,7 @@ std::optional<uint32_t> ParseKeyPathNum(std::span<const char> elem, bool& apostr
                 return false;
             }
 
+            substitutes.emplace();
```

Comment:
this implicit empty constructor was always weird to me, a more explicit and uniform alternative could be:
```suggestion
            substitutes = {};
```

Reply:
To me `.emplace()` appears more specific. Assigning `{}` seems to make it `std::nullopt`, leading to test failures.

Reply:
My mistake, thanks for checking